internal/datastore/src/letter-repository.ts

```diff
@@ -66,9 +70,7 @@ export class LetterRepository {
     return LetterSchema.parse(letterDb);
   }
 
-  async putLetterBatch(
-    letters: Omit<Letter, "ttl" | "supplierStatus" | "supplierStatusSk">[],
-  ): Promise<void> {
+  async putLetterBatch(letters: InsertLetter[]): Promise<void> {
```

Comment:
I would rename this to make sure it's not used accidentally. At the moment it's only used in the test data generation, but it doesn't implement the same existing ID check as the putLetter function.

```suggestion
  async unsafePutLetterBatch(letters: InsertLetter[]): Promise<void> {
```